api/artChainGlobalAPI.js

```diff
@@ -200,8 +200,43 @@ function ACGChainAPI() {
         return artwork_id;
     }
 
-    function buy_artwork(buyer_address, owner_address, artwork_id, artwork_prize) {
+    function buy_artwork(buyer_address, owner_address, artwork_id, artwork_prize, status) {
```

Comment:
- We don't need to add status argument. We have recorded the status of the artwork in the contract instead. 
- All the judgement regarding to the owner, the bidder address the bid number, have also been checked in the contract so we don't need to do it externally.
- To establish the safe transaction, we should call approveAndCall method instead of receiveApproval.